packages/workflow/src/index.ts

```diff
@@ -20,18 +20,19 @@
  * <!--SNIPSTART typescript-schedule-activity-workflow-->
  * <!--SNIPEND-->
  *
- * ### Signals and Queries
+ * ### Updates, Signals and Queries
  *
- * To add signal handlers to a Workflow, add a signals property to the exported `workflow` object. Signal handlers can
- * return either `void` or `Promise<void>`, you may schedule activities and timers from a signal handler.
+ * Use {@link setHandler} to set handlers for Updates, Signals, and Queries.
  *
- * To add query handlers to a Workflow, add a queries property to the exported `workflow` object. Query handlers must
- * **not** mutate any variables or generate any commands (like Activities or Timers), they run synchronously and thus
- * **must** return a `Promise`.
+ * Update and Signal handlers can be functions or async functions, and you may schedule activities, child workflows, timers etc.
```

Comment:
> `Signal handlers can be functions or async functions`

Sounds weird to me... "functions" encompass "async fonctions"... Maybe "can be either async or non-async functions"?

Reply:
> you may schedule activities, child workflows, timers etc

I agree this ought to be said clearly, but we should also call out that doing so require cautious handling. The SDK itself provides no guarantee that pending async operations have completed before the workflow execution itself completes.

Reply:
Changed as follows:
```
 * Update and Signal handlers can be either async or non-async functions. Update handlers may return a value, but signal
 * handlers may not (return `void` or `Promise<void>`). You may use Activities, Timers, child Workflows, etc in Update
 * and Signal handlers, but this should be done cautiously: for example, note that if you await async operations such as
 * these in an Update or Signal handler, then you are responsible for ensuring that the workflow does not complete first.
 *
 * Query handlers may **not** be async functions, and may **not** mutate any variables or use Activities, Timers,
 * child Workflows, etc.
```